packages/pwa-buildpack/lib/cli/generate-build-report.js

```diff
@@ -0,0 +1,213 @@
+        const grepBuffer = spawnSync('grep', ['Found'], {
+            input: whyBuffer.stdout
+        });
+
+        // [ 'info \r=> Found "@magento/pwa-buildpack@7.0.0"', '' ]
+        const outputArray = grepBuffer.stdout.toString().split('\n');
+
+        // [ '7.0.0' ]
+        const parsedOutputArray = outputArray
+            .filter(output => output.length > 0)
+            .map(output => output.split('@')[2].replace('"', ''));
+
+        dependencyMap.set(packageName, parsedOutputArray.toString());
+    });
+
+    return dependencyMap;
+}
+
+function getNpmDependencies(dependencies) {
+    const dependencyMap = new Map();
+
+    dependencies.map(packageName => {
+        const listbuffer = spawnSync('npm', ['list', packageName]);
+        const grepBuffer = spawnSync('grep', ['└──'], {
+            input: listbuffer.stdout
+        });
+
+        // TODO: handle multiple versions installed when not dupe. ie "npm ls core-js"
+        const [, , version] = grepBuffer.stdout
+            .toString()
+            .split('\n') // ["│ └── @magento/upward-js@5.0.0  deduped", "└── @magento/upward-js@5.0.0"]
+            .filter(text => !text.includes('deduped'))[0] // ["└── @magento/upward-js@5.0.0"]
+            .split('@'); // [ '└── ', 'magento/upward-js', '5.0.0' ]
+
+        dependencyMap.set(packageName, version);
+    });
+
+    return dependencyMap;
+}
+
+async function inspectBackend() {
+    prettyLogger.info('Inspecting Magento Backend');
+    const [projectConfig, sampleBackends] = await Promise.all([
+        loadEnvironment(
+            // Load .env from root
+            path.resolve(cwd)
+        ),
+        fetchSampleBackendUrls()
+    ]);
+
+    if (projectConfig.error) {
+        throw projectConfig.error;
+    }
+
+    const {
+        env: { MAGENTO_BACKEND_URL }
+    } = projectConfig;
+
+    if (sampleBackends.includes(MAGENTO_BACKEND_URL)) {
+        prettyLogger.log('Using sample backend: ', MAGENTO_BACKEND_URL);
+    } else {
+        prettyLogger.log('Not using sample backend.');
+    }
+
+    // Upcheck
+    try {
+        await fetchWithAgent(MAGENTO_BACKEND_URL);
+        prettyLogger.log('Backend is UP!');
+    } catch (e) {
+        prettyLogger.log('Backend is DOWN!');
+        prettyLogger.error('Reason:', e);
+    }
+
+    // Backend Version
+    // TODO: Implement if/when some endpoint other than magento_version exposes the version.
+    // try {
+    //     const res = await fetchWithAgent(
+    //         `${MAGENTO_BACKEND_URL}/magento_version`
+    //     );
+    //     prettyLogger.log('Magento Version:', await res.text());
+    // } catch (e) {
+    //     prettyLogger.warn(
+    //         'Unable to determine Magento version - ensure that the Magento_Version route is enabled when generating this report.'
+    //     );
+    //     prettyLogger.error('Reason:', e);
+    // }
+}
+
+function inspectBuildEnv() {
+    const osVersion = os.version();
```

Comment:
We need to be careful of this. `os.version` is added in node 13.

Failed in my local machine since I was running node 10. Do we recommend a minimum version of node for using PWA Studio?

![image](https://user-images.githubusercontent.com/35203638/102822461-bb3a8380-439e-11eb-80bb-5aa6cf0ce42e.png)

Reply:
Oops, you're right! Our monorepo [package.json](https://github.com/magento/pwa-studio/blob/develop/package.json#L88-L91) requires at least 10, but if this doesn't work in 10 we need a fall back (upgrading to minimum 12 is definitely out of scope).